File: financial-assistant-ui/src/app/models/plan/create/financial-goal.ts
export class FinancialGoal {
  constructor(name: string, amountOfMoney?: any, achieveAge?: any) {
    this.name = name;
    this.amountOfMoney = amountOfMoney;
    this.achieveAge = achieveAge;
  }

  public name: string;
  public amountOfMoney?: any;
  public achieveAge?: any;
}

File: plan-api/src/main/java/com/kpi/dyploma/planapi/model/plan/FinancialGoal.java
package com.kpi.dyploma.planapi.model.plan;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;
import lombok.ToString;

@Data
@Entity
@Table(name = "financial_goals")
public class FinancialGoal {
    @Id
    @GeneratedValue
    private Integer id;
    @ManyToOne(fetch= FetchType.LAZY, cascade=CascadeType.ALL)
    @JsonBackReference
    @ToString.Exclude
    private FinancialPlan financialPlan;
    private String name;
    private Integer amountOfMoney;
    private Integer achieveAge;
}
